src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,426 @@
+    time_cut_outliers : List[2]
+        Interval (in waveform samples) of accepted time values
+    sigma_clipping_max_sigma : int
+        Maximal value for the sigma clipping outlier removal
+    sigma_clipping_iterations : int
+        Number of iterations for the sigma clipping outlier removal
+    """
+
+    sample_size = IntTelescopeParameter(2500, help="sample size").tag(config=True)
+    update_frequency = IntTelescopeParameter(2500, help="update frequency").tag(
+        config=True
+    )
+    image_extractor_type = TelescopeParameter(
+        trait=ComponentName(ImageExtractor, default_value="LocalPeakWindowSum"),
+        default_value="LocalPeakWindowSum",
+        help="Name of the ImageExtractor subclass to be used.",
+    ).tag(config=True)
+    charge_median_cut_outliers = List(
+        [-0.3, 0.3],
+        help="Interval of accepted charge values (fraction with respect to camera median value)",
+    ).tag(config=True)
+    charge_std_cut_outliers = List(
+        [-3, 3],
+        help="Interval (number of std) of accepted charge standard deviation around camera median value",
+    ).tag(config=True)
+    time_cut_outliers = List(
+        [0, 60], help="Interval (in waveform samples) of accepted time values"
+    ).tag(config=True)
+    sigma_clipping_max_sigma = Int(
+        default_value=4,
+        help="Maximal value for the sigma clipping outlier removal",
+    ).tag(config=True)
+    sigma_clipping_iterations = Int(
+        default_value=5,
+        help="Number of iterations for the sigma clipping outlier removal",
+    ).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        """
+        Parameters
+        ----------
+        subarray: ctapipe.instrument.SubarrayDescription
+            Description of the subarray
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        kwargs
+        """
+        super().__init__(config=config, parent=parent, subarray=subarray, **kwargs)
+
+        # Check validity of sample size and update frequency
+        if self.sample_size < self.update_frequency:
+            raise ValueError(
+                f"The 'update_frequency' ({self.update_frequency}) must be less than or equal to the 'sample_size' ({self.sample_size})."
+            )
+        # set up the buffer
+        self.n_events_in_buffer = 0  # number of events in sample
+        self.trigger_time = []  # trigger time of event in sample
+        self.charges = []  # charge per event in sample
+        self.arrival_times = []  # arrival time per event in sample
+        self.broken_pixels = []  # masked pixels per event in sample
+
+        # load the waveform charge extractor
+        self.extractor = ImageExtractor.from_name(
```

Comment:
Since you have `image_extractor` as a `TelescopeParameter`, you need to create all types requested, not just the one type.

See e.g.:
https://github.com/cta-observatory/ctapipe/blob/eb43957c838bfe30fd1a6403601b724b984a25e4/src/ctapipe/calib/camera/calibrator.py#L130